core/src/loader/fileurl.rs

```diff
@@ -0,0 +1,128 @@
+use super::{Loader, RemoteDocument};
+use futures::future::{BoxFuture, FutureExt};
+use locspan::Meta;
+use rdf_types::{vocabulary::IriIndex, IriVocabulary};
+use std::fmt;
+use std::fs::File;
+use std::io::{BufReader, Read};
+use url::Url;
+
+/// File-URL loader.
+///
+/// This is a special JSON-LD document loader that can load document from any file: URL.
+///
+/// Loaded documents are not cached: a new file system read is made each time
+/// an URL is loaded even if it has already been queried before.
+pub struct FileUrlLoader<
+	I = IriIndex,
+	M = locspan::Location<I>,
+	T = json_ld_syntax::Value<M>,
+	E = json_ld_syntax::parse::MetaError<M>,
+> {
+	parser: Box<DynParser<I, M, T, E>>,
+}
+
+impl<I: Send, T: Send, M: Send, E> Loader<I, M> for FileUrlLoader<I, M, T, E> {
+	type Output = T;
+	type Error = Error<E>;
+
+	fn load_with<'a>(
+		&'a mut self,
+		vocabulary: &'a mut (impl Sync + Send + IriVocabulary<Iri = I>),
+		url: I,
+	) -> BoxFuture<'a, Result<RemoteDocument<I, M, T>, Self::Error>>
+	where
+		I: 'a,
+	{
+		async move {
+			let url_str = vocabulary.iri(&url).unwrap().as_str();
+			if !url_str.starts_with("file:") {
+				return Err(Error::NotFileUrl(url_str.into()));
+			}
+			let url_parsed = Url::parse(url_str).map_err(Error::InvalidUrl)?;
```

Comment:
You can use [`Iri::as_uri`](https://docs.rs/iref/latest/iref/iri/struct.Iri.html#method.as_uri) for that.